darts/dataprocessing/transformers/boxcox.py

```diff
@@ -0,0 +1,102 @@
+"""
+Box-Cox Transformer
+-------------------
+"""
+
+from typing import Optional, Union, Sequence
+from scipy.stats import boxcox_normmax, boxcox
+from scipy.special import inv_boxcox
+
+from darts.timeseries import TimeSeries
+from darts.dataprocessing import Validator
+from darts.dataprocessing.transformers import FittableDataTransformer, InvertibleDataTransformer
+from darts.logging import get_logger, raise_if
+
+logger = get_logger(__name__)
+
+
+class BoxCox(FittableDataTransformer[TimeSeries], InvertibleDataTransformer[TimeSeries]):
+
+    def __init__(self,
+                 name: str = "BoxCox",
+                 validators: Optional[Sequence[Validator]] = None):
+        """
+        Box-Cox data transformer.
+        See https://otexts.com/fpp2/transformations.html#mathematical-transformations for more information
+
+        Parameters
+        ----------
+        name
+            A specific name for the transformer
+        validators
+            Sequence of validators that will be called before transform()
+        """
+        super().__init__(name=name, validators=validators)
+        self._lmbda = None
+
+    def fit(self,
+            data: TimeSeries,
+            lmbda: Optional[Union[float, Sequence[float]]] = None,
+            optim_method='mle') -> 'BoxCox':
+        """
+        Sets the `lmbda` parameter value.
+
+        Parameters
+        ----------
+        data
+            The time series to fit on
+        lmbda
+            If None given, will automatically find an optimal value of lmbda (for each dimension
+            of the time series) using `scipy.stats.boxcox_normmax` with `method=optim_method`
+            If a single float is given, the same lmbda value will be used for all dimensions of the series.
+            Also allows to specify a different lmbda value for each dimension of the time series by passing
+            a sequence of values.
+
+        Returns
+        -------
+            Fitted transformer (self)
+        """
+        super().fit(data)
+
+        raise_if(not isinstance(optim_method, str) or optim_method not in ['mle', 'pearsonr'],
+                 "optim_method parameter must be either 'mle' or 'pearsonr'",
+                 logger)
+
+        if lmbda is None:
+            # Compute optimal lmbda for each dimension of the time series
+            lmbda = data.pd_dataframe().apply(boxcox_normmax, method=optim_method)
```

Comment:
Not much to do with this PR, but should we plan to have an option in `pd_dataframe()` to avoid copying the dataframe, for those internal use cases where you really know what you are doing? That would improve performance.

Reply:
I actually thought about this when writing these lines… I think we can already do this, by accessing `series._df`.

The reason I went with `pd_dataframe()` is because I wasn't actually working from within the `TimeSeries` class, but if you think that it's okay to access the dataframe directly from here, I also agree. I'll change it.